Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -443,47 +492,68 @@ def print_latest_version(info, verbose:, ambiguous_cask: false)
         info[:version][:latest]
       end
 
-      puts "#{formula_or_cask_s}: #{current_s} ==> #{latest_s}"
+      puts "#{package_or_resource_s}: #{current_s} ==> #{latest_s}"
+    end
+
+    # Prints the livecheck result for a resources for a given Formula.
+    sig { params(info: T::Array[Hash], verbose: T::Boolean).void }
+    def print_resources_info(info, verbose: false)
+      info.each do |r_info|
+        if r_info[:status] && r_info[:messages]
+          SkipConditions.print_skip_information(r_info)
+        else
+          print_latest_version(
+            r_info,
+            verbose:  verbose,
+            resource: true,
+          )
+        end
+      end
     end
 
     sig {
       params(
-        livecheck_url:   T.any(String, Symbol),
-        formula_or_cask: T.any(Formula, Cask::Cask),
+        livecheck_url:       T.any(String, Symbol),
+        package_or_resource: T.any(Formula, Cask::Cask, Resource),
       ).returns(T.nilable(String))
     }
-    def livecheck_url_to_string(livecheck_url, formula_or_cask)
+    def livecheck_url_to_string(livecheck_url, package_or_resource)
       case livecheck_url
       when String
         livecheck_url
       when :url
-        formula_or_cask.url&.to_s if formula_or_cask.is_a?(Cask::Cask)
+        package_or_resource.url&.to_s if package_or_resource.is_a?(Cask::Cask) || package_or_resource.is_a?(Resource)
       when :head, :stable
-        formula_or_cask.send(livecheck_url)&.url if formula_or_cask.is_a?(Formula)
+        # Since resource's "url" is considered "stable" by default.
+        # And some resources may contain in "head" block as well
+        package_or_resource.send(:url)&.to_s if package_or_resource.is_a?(Resource)
```

Comment:
> I asked @nandahkrishna once that :head and :stable are implemented in Formula as a SoftwareSpec. And (to my knowledge) if we want to have the same for resources, we have to update the Resource DSL. And he told me:
> 
> > Resource's :url is considered :stable by default.
>
> And I took it in a sense that if I got :stable symbol, I can return :url for resource in #livecheck_url_to_string method. (kindly do correct me if I am mistaken --- which I think I most probably am)
>
> Anyways, I have removed tests for :head and :stable for the time being.

I'm of the opinion that `url :head` and `url :stable` in resource `livecheck` blocks should work the same as casks, where `#livecheck_url_to_string` doesn't handle these symbols (i.e., implicitly returning `nil`). At least at present, neither casks nor resources respond to `#head` or `#stable` methods and they're not part of their DSLs, so using those values in the `livecheck` block shouldn't work.

At least to me, we shouldn't be trying to quietly correct this issue for users by substituting the resource `url`, as this will lead people to think that `url :head` and `url :stable` are valid and we don't want this mistake to propagate through homebrew/core. Ideally, we should enforce valid symbols with a Rubocop but that's something for a later PR because currently the related Rubocops only work for formulae (not casks or resources) and fixing that is a significant undertaking. [I have some in-progress work but it's currently blocked while I wait for #13233 to be finished/merged.]

That said, at least from my perspective, this comment and statement should be removed from `#livecheck_url_to_string` (so the method will implicitly return `nil` when these symbols are used in a resource `livecheck` block, same as casks).

@nandahkrishna What do you think about this?

Reply:
After some thought, I agree with Sam on this – let's remove this comment and statement for consistency with the Cask behaviour. So in short, we will get the URL only if we use `:url`, and not `:stable` or `:head`.

I was earlier concerned about `:stable` being valid in some sense (resources included in a formula are most likely "stable" by default), but the explanation that resources do not respond to the `head` and `stable` methods convinces me that we shouldn't quietly succeed by returning the `url`.

Reply:
Alright! I have removed this. Thank you @samford and @nandahkrishna for the explanation.